R/pca.R

```diff
@@ -127,7 +127,7 @@ ui_g_pca <- function(id,
               width = 12,
               div(
                 class = "my-5",
-                plotOutput(ns("plot_pca"))
+                teal.widgets::plot_with_settings_ui(ns("plot_pca"))
               ),
               DT::DTOutput(ns("table_pca"))
```

Comment:
@Polkas one question, how does the `table_with_settings()` work, do we have an example from another teal module? we tried to use here but gave an error, and I just tried to find another teal module but did not find any for some reason, thanks

Reply:
I see the documentation for table_with_settings is very poor. It looks like only rtables object could be used but you want to pass a regular data.frame to `teal.widgets::table_with_settings_srv`.

Almost all tmc modules for tables (tm_t_*) are returning rtables and printing it with table with settings.
e.g. https://github.com/insightsengineering/teal.modules.clinical/blob/main/R/tm_t_events.R

Reply:
a new issue https://github.com/insightsengineering/teal.widgets/issues/101

Reply:
ok thanks @Polkas then we will keep the code as is here - the `DT` is nice here for quick exploration of the data frame, so we don't want to revert that to `rtables`